### ISSUE TITLE
`python39Packages.scipy` build failure on x86_64-linux as of `2da37482`

### ISSUE THREAD

**Issue by samuela** [NixOS/nixpkgs#210797]
## Issue description
Build of `python39Packages.scipy` failed on x86_64-linux as of 2da37482b1c5d99851dc683c714429734cba4191. This is currently breaking `python39Packages.tensorflow-bin`.

```
error: builder for '/nix/store/wn2dvlr1nxyw7f91f300yc7iv48lhlin-python3.9-scipy-1.9.3.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
       >   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "/nix/store/7smx7z01rr4rc8bkl792rz1s622v0qxg-python3.9-meson-python-0.12.0/lib/python3.9/site-packages/mesonpy/__init__.py", line 45, in <module>
       >     import mesonpy._compat
       >   File "/nix/store/7smx7z01rr4rc8bkl792rz1s622v0qxg-python3.9-meson-python-0.12.0/lib/python3.9/site-packages/mesonpy/_compat.py", line 15, in <module>
       >     from typing_extensions import ParamSpec
       > ModuleNotFoundError: No module named 'typing_extensions'
       > 
       > 
       For full logs, run 'nix log /nix/store/wn2dvlr1nxyw7f91f300yc7iv48lhlin-python3.9-scipy-1.9.3.drv'.
error: 1 dependencies of derivation '/nix/store/r89zhz01v18dp4sjlsmigillz5chdalk-python3.9-tensorflow-gpu-2.9.3.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/3920896263)

python39Packages.tensorflow-bin maintainers cc: @jyp @abbradar @cdepillabout
python39Packages.scipy maintainers cc: @FRidh
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 2da37482b1c5d99851dc683c714429734cba4191
2. Run `nix-build -A python39Packages.scipy`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-1030-azure, Ubuntu, 22.04.1 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - nixpkgs: `/nix/store/85kimkz0rlm1r7pv3w04sllbad0dcbkp-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/3920896263) that created this issue.
- Internal tags: 6473183954746190696 -6240217109778611740

**Comment by samuela** [comment]
Not entirely sure what to make of as the cause of this issue. The last change to `scipy` was Nov 18, 2022: https://github.com/NixOS/nixpkgs/commit/1b3d8a0dc893d5382f00958bb0f7407b71bb68cf. Perhaps it was a change in `mesonpy`?

**Comment by risicle** [comment]
Looks like this might have just been addressed in #210878. Please reopen if not.
